subgraphs/lido/src/entityUpdates/financialMetrics.ts

```diff
@@ -0,0 +1,352 @@
+  financialMetrics.dailyTotalRevenueUSD =
+    poolMetricsDailySnapshot.dailyTotalRevenueUSD;
+  financialMetrics.save();
+}
+
+export function updateProtocolSideRevenueMetrics(
+  block: ethereum.Block,
+  amount: BigInt
+): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const protocol = getOrCreateProtocol();
+  const financialMetrics = getOrCreateFinancialDailyMetrics(block);
+  const poolMetricsDailySnapshot = getOrCreatePoolsDailySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+  const poolMetricsHourlySnapshot = getOrCreatePoolsHourlySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+
+  // TODO: Rewards are minted in stETH shares, should we price with stETH? CONFIRM.
+  const amountUSD = bigIntToBigDecimal(amount).times(
+    getOrCreateToken(Address.fromString(ETH_ADDRESS), block.number)
+      .lastPriceUSD!
+  );
+
+  // Pool
+  pool.cumulativeProtocolSideRevenueUSD = pool.cumulativeProtocolSideRevenueUSD.plus(
+    amountUSD
+  );
+  pool.save();
+
+  // Pool Daily
+  poolMetricsDailySnapshot.cumulativeProtocolSideRevenueUSD =
+    pool.cumulativeProtocolSideRevenueUSD;
+  poolMetricsDailySnapshot.dailyProtocolSideRevenueUSD = poolMetricsDailySnapshot.dailyProtocolSideRevenueUSD.plus(
+    amountUSD
+  );
+  poolMetricsDailySnapshot.save();
+
+  // Pool Hourly
+  poolMetricsHourlySnapshot.cumulativeProtocolSideRevenueUSD =
+    pool.cumulativeProtocolSideRevenueUSD;
+  poolMetricsHourlySnapshot.hourlyProtocolSideRevenueUSD = poolMetricsHourlySnapshot.hourlyProtocolSideRevenueUSD.plus(
+    amountUSD
+  );
+  poolMetricsHourlySnapshot.save();
+
+  // Protocol
+  protocol.cumulativeProtocolSideRevenueUSD =
+    pool.cumulativeProtocolSideRevenueUSD;
+  protocol.save();
+
+  // Financial Daily
+  financialMetrics.cumulativeProtocolSideRevenueUSD =
+    pool.cumulativeProtocolSideRevenueUSD;
+  financialMetrics.dailyProtocolSideRevenueUSD = financialMetrics.dailyProtocolSideRevenueUSD.plus(
+    amountUSD
+  );
+  financialMetrics.save();
+}
+
+export function updateSupplySideRevenueMetrics(block: ethereum.Block): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const protocol = getOrCreateProtocol();
+  const financialMetrics = getOrCreateFinancialDailyMetrics(block);
+  const poolMetricsDailySnapshot = getOrCreatePoolsDailySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+  const poolMetricsHourlySnapshot = getOrCreatePoolsHourlySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+
+  // Pool
+  pool.cumulativeSupplySideRevenueUSD = pool.cumulativeTotalRevenueUSD.minus(
+    pool.cumulativeProtocolSideRevenueUSD
+  );
+  pool.save();
+
+  // Pool Daily
+  poolMetricsDailySnapshot.cumulativeSupplySideRevenueUSD =
+    pool.cumulativeSupplySideRevenueUSD;
+  poolMetricsDailySnapshot.dailySupplySideRevenueUSD = poolMetricsDailySnapshot.dailyTotalRevenueUSD.minus(
```

Comment:
Some of the values in `dailySupplySideRevenueUsd` is zero ([Link](https://subgraphs.messari.io/subgraph?endpoint=https%3A%2F%2Fapi.thegraph.com%2Fsubgraphs%2Fname%2Ffortysevenlabs%2Flido&tab=pool&poolId=0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84&view=poolDailySnapshots-dailySupplySideRevenueUSD)). Can you add a condition here to only update the field when the revenue is positive?

Reply:
Pushed the change.